### PR TITLE
test using macos-14 runner instead of macos-12

### PR DESCRIPTION
**Pull request by ggreenway**
Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by ggreenway** [comment]
@phlax not sure if this will work as I'm hoping or not, but want to see what happens if we run the mac CI on a later version. Related to the CEL build failures that weren't detected in CI.

**Comment by phlax** [comment]
> not sure if this will work as I'm hoping or not

the change wont be triggered in a PR - ill test it now in other repo ...

File: .github/workflows/envoy-macos.yml [hunk 1]
```diff
@@ -48,7 +48,7 @@ jobs:
       command:
       container-command:
       request: ${{ needs.load.outputs.request }}
-      runs-on: macos-12-xl
+      runs-on: macos-14-xl
```
Comment (phlax) [inline .github/workflows/envoy-macos.yml:51]:
```suggestion
      runs-on: macos-14-xlarge
```

trying to make sense of docs - which are at least incomplete (eg `macos-12-xl` is mentioned elsewhere and works) here:

https://docs.github.com/en/actions/using-github-hosted-runners/about-larger-runners/running-jobs-on-larger-runners?platform=mac#available-macos-larger-runners

one thing im noticing is that `large` <> `xlarge` seem to have their names in reverse (large is "larger" than x-large!!!) 

also seems to relate more to cpu type

anyhow testing elsewhere and `macos-14-xlarge` works insofar as it runs - ci didnt fail yet - but taking some time

Comment (phlax) [reply .github/workflows/envoy-macos.yml:51]: ci passed - @ggreenway ill post you the link offline

**Review by phlax — APPROVED** [review]
lgtm, thanks @ggreenway

**Pull request merged**